### PR TITLE
Fix Bash Completion

### PR DESCRIPTION
**Pull request by benhamill**
Behold:

```
● ~ ❱ source ~/bash_completion.d/docker-compose
-sh: `__docker-compose_compose_file': not a valid identifier
-sh: `___docker-compose_all_services_in_compose_file': not a valid identifier
-sh: `__docker-compose_services_all': not a valid identifier
-sh: `___docker-compose_services_with_key': not a valid identifier
-sh: `__docker-compose_services_from_build': not a valid identifier
-sh: `__docker-compose_services_from_image': not a valid identifier
-sh: `__docker-compose_services_with': not a valid identifier
-sh: `__docker-compose_services_running': not a valid identifier
-sh: `__docker-compose_services_stopped': not a valid identifier
-sh: `_docker-compose_build': not a valid identifier
-sh: `_docker-compose_docker-compose': not a valid identifier
-sh: `_docker-compose_help': not a valid identifier
-sh: `_docker-compose_kill': not a valid identifier
-sh: `_docker-compose_logs': not a valid identifier
-sh: `_docker-compose_port': not a valid identifier
-sh: `_docker-compose_ps': not a valid identifier
-sh: `_docker-compose_pull': not a valid identifier
-sh: `_docker-compose_restart': not a valid identifier
-sh: `_docker-compose_rm': not a valid identifier
-sh: `_docker-compose_run': not a valid identifier
-sh: `_docker-compose_scale': not a valid identifier
-sh: `_docker-compose_start': not a valid identifier
-sh: `_docker-compose_stop': not a valid identifier
-sh: `_docker-compose_up': not a valid identifier
-sh: `_docker-compose': not a valid identifier
```

Seems really weird. I assume it's something wrong with either the bash completion file or my shell. Anyone got any pointers?

```
● ~ ❱ docker-compose --version
docker-compose 1.1.0
```

### CODE REVIEW

**Comment by benhamill** [comment]
I just added a change to this that I noticed after upgrading to 1.6.0, which is that there was a string that was a regex, but it wasn't quoted and was giving invalid syntax input when this file is sourced.

@pnathan: How does this look to you?

**Comment by benhamill** [comment]
I rebased this sucker. Someone please have a look?

**Comment by thaJeztah** [comment]
thanks!

ping @albers Harald, if you have the chance :-)

**Comment by aanand** [comment]
LGTM

File: contrib/completion/bash/docker-compose [hunk 1]
```diff
@@ -131,25 +131,25 @@ _docker-compose_kill() {
...
 		*)
-			__docker-compose_services_all
+			__docker_compose_services_all
 			;;
 	esac
 }
 
 
-_docker-compose_migrate-to-labels() {
+_docker_compose_migrate_to_labels() {
```
Comment (albers) [inline contrib/completion/bash/docker-compose]: As `migrate-to-labels` contains dashes, completion for it does not work. The generic approach for identifying subcommands expects `_docker_compose_migrate-to-labels()`. An additional transformation from `-` to `_` is needed in the subcommand parsing routine.

Please change
```
local completions_func=_docker_compose_${command}
```
to
```
local completions_func=_docker_compose_${command//-/_}
```

Comment (benhamill) [inline contrib/completion/bash/docker-compose]: Gah! Good eye. Without tests for this stuff (I wouldn't even know how), that's easy to miss. I'll fix it and push again tomorrow. Yay!

**Comment by benhamill** [comment]
@albers: I revised my commit to catche dashes in subcommands (like `migrate-to-labels`).

What do y'all think, now?

**Comment by albers** [comment]
I still don't understand why you are experiencing these problems here. After fixing #1386 (which was about bash 3.2.57 on OS X 10.10.3.), there seemed to be no more problems with macs any more.

@benhamill can you please give the exact OS and bash versions you use? Did you use the recommended installation procedure from the Compose docs?

@pnathan the issue your reported is a different one: `_get_comp_words_by_ref` is a function from the bash-completion package. Obviously this must be installed (in a recent version) for completion to work. This is not OS specific.
We definitely need installer packages for Compose. Manual installation is too error prone.

Anyway, using dashes in function names looks like danger seeking to me. Your fix is definitely an improvement. Thanks for it, Ben!

LGTM

**Comment by aanand** [comment]
LGTM

**Comment by benhamill** [comment]
Folks around me who've seen this, including me, are installing Compose from homebrew. My understanding is that this is a very, very common way to install Compose. And I've seen it on OS X versions 10.9 and 10.10.

As for bash:
```
● ~ ❱ bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin14)
Copyright (C) 2007 Free Software Foundation, Inc.
```

Hope that helps. Would love to see this merged soon. :heart:

**Comment by mnowster** [comment]
LGTM

**Comment by benhamill** [comment]
Yaaaaaaaaaay! :sparkles:Thanks!:sparkles:

**Comment by albers** [comment]
Just for the records: Bash supports dashes in function names unless it is executed in posix mode:
```bash
$ bash /usr/share/bash-completion/completions/docker
$ bash --posix /usr/share/bash-completion/completions/docker
/usr/share/bash-completion/completions/docker: line 670: `__docker_complete_detach-keys': not a valid identifier
```

**Pull request merged**